packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -11,53 +19,102 @@ interface VersionData {
   latest: Version
 }
 
+const REMOTE_MANIFEST_URL = 'https://download.cypress.io/desktop.json'
+const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
+
+type GetLatestVersionOptions = { initialLaunch: boolean, testingType: TestingType | null, currentCypressVersion: string }
+
 export class VersionsDataSource {
-  static result: undefined | Promise<string>
+  constructor (private ctx: DataContext) {}
+
+  static npmMetadata: undefined | Record<string, string>
+  static initialLaunch: boolean = true
 
   /**
    * Returns most recent and current version of Cypress
    * {
    *   current: {
+   *     id: '8.7.0',
    *     version: '8.7.0',
    *     released: '2021-10-15T21:38:59.983Z'
    *   },
    *   latest: {
+   *     id: '8.8.0',
    *     version: '8.8.0',
    *     released: '2021-10-25T21:38:59.983Z'
    *   }
    * }
    */
   async versions (): Promise<VersionData> {
-    const currentCypressVersion = require('@packages/root')
+    const currentCypressVersion = pkg.version
+    const latestVersion = await this.getLatestVersion({ initialLaunch: VersionsDataSource.initialLaunch, testingType: this.ctx.coreData.currentTestingType, currentCypressVersion })
 
-    VersionsDataSource.result ??= execa(`npm`, [`view`, `cypress`, `time`, `--json`]).then((result) => result.stdout)
+    VersionsDataSource.initialLaunch = false
+    VersionsDataSource.npmMetadata ??= await this.getVersionMetadata()
 
-    const result = await VersionsDataSource.result
+    const latestVersionMetadata: Version = {
+      id: latestVersion,
+      version: latestVersion,
+      released: VersionsDataSource.npmMetadata[latestVersion] ?? new Date().toISOString(),
+    }
 
-    const json = JSON.parse(result)
+    return {
+      latest: latestVersionMetadata,
+      current: {
+        id: currentCypressVersion,
+        version: currentCypressVersion,
+        released: VersionsDataSource.npmMetadata[currentCypressVersion] ?? new Date().toISOString(),
+      },
+    }
+  }
+
+  private async getVersionMetadata (): Promise<Record<string, string>> {
+    const response = await this.ctx.util.fetch(NPM_CYPRESS_REGISTRY)
+    const responseJson = await response.json()
 
-    delete json['modified']
-    delete json['created']
+    debug('NPM release dates %o', responseJson.time)
 
-    const latest = Object.keys(json).sort().reverse()?.[0]
+    return responseJson.time
+  }
 
-    if (!latest) {
-      throw Error('Could not get npm info for Cypress')
+  private async getLatestVersion ({ initialLaunch, testingType, currentCypressVersion }: GetLatestVersionOptions): Promise<string> {
```

Comment:
so now that this is here, can we kill off `packages/server/lib/updater.js`? https://github.com/cypress-io/cypress/blob/21ffca602998159869a514c3b802658e12d23846/packages/server/lib/updater.js#L9-L26

Reply:
Removed: https://github.com/cypress-io/cypress/commit/71933ef5a6a332e30348bbead924caf4749061fc